### PR TITLE
support private ACL for object

### PR DESCRIPTION
**Pull request by lelikg**
support list of ACL for object

### CODE REVIEW

File: fakestorage/object_test.go [hunk 1]
```diff
@@ -604,3 +604,33 @@ func TestServerClientObjectDeleteErrors(t *testing.T) {
...
+
+func TestServerClientObjectSetAclPrivate(t *testing.T) {
+	objs := []Object{
+		{BucketName: "some-bucket", Name: "img/public-to-private.jpg"},
+	}
+
+	runServersTest(t, objs, func(t *testing.T, server *Server) {
+		t.Run("public to private", func(t *testing.T) {
+			ctx := context.Background()
+			objHandle := server.Client().Bucket("some-bucket").Object("img/public-to-private.jpg")
```
Comment (alon-ne) [inline fakestorage/object_test.go:616]: Here you get a handle to this object, but who created it?

File: fakestorage/object_test.go [hunk 2]
```diff
@@ -604,3 +604,33 @@ func TestServerClientObjectDeleteErrors(t *testing.T) {
...
+				return
+			}
+
+			rules, err := objHandle.ACL().List(ctx)
+			if err != nil {
+				t.Errorf("unexpected error while getting acl %+v", err)
+				return
+			}
+
+			if len(rules) != 1 && rules[0].Entity != storage.AllAuthenticatedUsers && rules[0].Role != storage.RoleReader {
```
Comment (alon-ne) [inline fakestorage/object_test.go]: && should be ||

Comment (alon-ne) [reply fakestorage/object_test.go]: Or better yet, make separate if's with different test failures

File: fakestorage/server.go [hunk 3]
```diff
@@ -155,6 +155,8 @@ func (s *Server) buildMuxer() {
 	r.Path("/b/{bucketName}").Methods("GET").HandlerFunc(s.getBucket)
 	r.Path("/b/{bucketName}/o").Methods("GET").HandlerFunc(s.listObjects)
 	r.Path("/b/{bucketName}/o").Methods("POST").HandlerFunc(s.insertObject)
+	r.Path("/b/{bucketName}/o/{objectName:.+}/acl").Methods("GET").HandlerFunc(s.listObjectAcl)
+	r.Path("/b/{bucketName}/o/{objectName:.+}/acl/{entity}").Methods("PUT").HandlerFunc(s.setObjectAcl)
```
Comment (alon-ne) [inline fakestorage/server.go]: Why is entity in the request path? I thought it's a json object?

**Review by alon-ne — CHANGES_REQUESTED** [review]

File: fakestorage/object_test.go (hunk 1 above)
Comment (lelikg) [reply fakestorage/object_test.go:616]: the runServersTest is an internal function that creates the objs that you pass to it

File: fakestorage/object_test.go (hunk 2 above)
Comment (lelikg) [reply fakestorage/object_test.go]: done

File: fakestorage/server.go (hunk 3 above)
Comment (lelikg) [reply fakestorage/server.go]: it's the way google client sends it

Comment (alon-ne) [reply fakestorage/server.go]: With the whole json object encoded in the url...?

**Review by alon-ne — APPROVED** [review]

**Review by idanya — APPROVED** [review]

**Pull request merged**